fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializationContext.java

```diff
@@ -370,8 +371,16 @@ public int compare(MaterializationContext o1, MaterializationContext o2) {
             OperatorType o2Type = o2.getMvExpression().getOp().getOpType();
 
             if (o1Type == o2Type && (o1Type == OperatorType.LOGICAL_AGGR)) {
+                boolean mvHasDifferentRows = orderingRowCount(o1) != 0 && orderingRowCount(o2) != 0
```

Comment:
is this policy too sensitive? for an exmaple, MV0 and MV1 have N rows and N+1 rows respectively,  MV1 may be a promising one for the certain query? so mvHasDifferentRows should introduce an epsilon(ε) so that |MV1-MV1|/max(|MV1|,|MV2|) > ε, then we consider that the larger one is superior to the smaller one significantly.

Reply:
But we don't know have enough clue to tell if MV1 is actually better than MV0 or not here.
if mvHasDifferentRows is set to `false` when MV0 and MV1 have `similar` row number, the later code still compares the dimension diff number and later the row number.

My point is that we can't exactlly tell which mv is better for a particular query, we can only believe that comparing row count is more promising than comparing dimension diff number.

Reply:
rowCount is promising only when there is a significant gap between two MVs' rowcounts.  if the row counts are  almost the same,   we also can not tell out which MV is better.  I show an exmaple:
1. MV0: [d0,d1,d2, sum(m0)];
2. MV1: [d1,d0,d3, sum(m0)];
we suppose that |MV0| = N,  |MV1|= M, N>M and N-M is small;

for query: select d1, sum(m0) from t where predicate(d0) group by d1;
if MV0 has short key on d0, this query fetches data from IO layer quickly.  
so I think only two MV's rowcount are different drastically, the one of smaller rowcount is promissing.  if they are close, we should not use the rowcounts.

Reply:
I understand your scenario and the limitation of exactly comparing mv row count.

What I do in this PR is prefering comparing mv rount to comparing the deminsion diff number, I get that this is not the most perfect solution. But it is much better than current logic.

Current we don't have enough information(eg: the key columns used in query's predicate and how many rows can be reduced by this predicate, etc) to make sure that MV1 is better choice than MV0 when MV1 has larger row count, other than not choosing MV0 when MV0 has less row count. 

My point is that we can't tell which mv is better when the two mv have similar row count according to the current information from the context at current stage, it's still better comparing row count thant comparing dimension diff number.

Reply:
I have changed the logic in `BestMvSeclector` which is the actually class choosing mv according to your comment